### PR TITLE
[M931] Select All Filter

### PR DESCRIPTION
**Pull request by tienhoah**

### CODE REVIEW

**Comment by tienhoah** [comment]
[method dropdown: all/none toggle](https://trello.com/c/pesVhx2Y/931-method-dropdown-all-none-toggle)

File: src/app/project/controllers/collect-records.ctrl.js [hunk 1]
```diff
@@ -34,10 +34,12 @@ angular.module('app.project').controller('CollectRecordsCtrl', [
     'use strict';
 
     var collectRecordsTable;
+    var collectRecordsCount;
     var addTransectGroupButton;
     var promises;
     var project_id = $stateParams.project_id;
 
+    $scope.someTestVal = 11;
```
Comment (saanobhaai) [inline src/app/project/controllers/collect-records.ctrl.js]: remove

File: src/app/project/controllers/collect-records.ctrl.js [hunk 2]
```diff
@@ -404,13 +447,30 @@ angular.module('app.project').controller('CollectRecordsCtrl', [
...
+        $scope.tableControl.records.length === collectRecordsCount
+      );
+    };
+
     promises = [
       authService.getCurrentUser(),
       offlineservice.CollectRecordsTable(project_id)
     ];
 
     $q.all(promises).then(function(output) {
```
Comment (saanobhaai) [inline src/app/project/controllers/collect-records.ctrl.js]: If these promises don't resolve, `collectRecordsCount` doesn't get set -- maybe it should be given a default on declaration

File: src/app/project/controllers/collect-records.ctrl.js [hunk 3]
```diff
@@ -57,12 +59,15 @@ angular.module('app.project').controller('CollectRecordsCtrl', [
       ProjectService.HABITAT_COMPLEXITY_TRANSECT_TYPE,
       ProjectService.BLEACHING_QC_QUADRAT_TYPE
     ];
-
     var statusChoices = [null, 'ok', 'warning', 'error'];
+    var protocolMethodsLength = protocolMethods.length;
```
Comment (saanobhaai) [inline src/app/project/controllers/collect-records.ctrl.js]: Not seeing the need for additional vars; these are only used in one place each. All comments on this file apply also to transect methods below.

File: src/app/project/controllers/collect-records.ctrl.js [hunk 4]
```diff
@@ -34,10 +34,12 @@ angular.module('app.project').controller('CollectRecordsCtrl', [
     'use strict';
 
     var collectRecordsTable;
+    var collectRecordsCount;
```
Comment (saanobhaai) [inline src/app/project/controllers/collect-records.ctrl.js]: Should we move to const/let?

File: src/app/project/controllers/collect-records.ctrl.js [hunk 5]
```diff
@@ -354,38 +369,66 @@ angular.module('app.project').controller('CollectRecordsCtrl', [
             utils.template(utils.messages.deleteRecordWarning, args)
           );
         },
-        filterMethod: function(showMethod, protocol) {
+        filterMethod: function(item) {
+          var selected = item.selected;
```
Comment (saanobhaai) [inline src/app/project/controllers/collect-records.ctrl.js]: Again not seeing need for additional vars

File: src/app/project/controllers/collect-records.ctrl.js [hunk 6]
```diff
@@ -404,13 +447,30 @@ angular.module('app.project').controller('CollectRecordsCtrl', [
       });
     };
 
+    $scope.tableControl.getFilteredRecordsCount = function() {
+      return (
+        $scope.tableControl.records &&
+        `${$scope.tableControl.records.length}/${collectRecordsCount}`
```
Comment (saanobhaai) [inline src/app/project/controllers/collect-records.ctrl.js]: What happens when `collectRecordsCount` is 0?

File: src/app/project/controllers/collect-records.ctrl.js [hunk 7]
```diff
@@ -466,6 +526,27 @@ angular.module('app.project').controller('CollectRecordsCtrl', [
...
 
+    function selectAllOptions(allSelected, filter_options) {
+      var filterTypes = filter_options.filterTypes;
+      var choices = filter_options.choices;
+      var storageName = filter_options.storageName;
+      var options = JSON.parse(localStorage.getItem(storageName)) || choices;
+
+      if (allSelected) {
+        filterTypes.forEach(filterItem => {
+          if (!filterItem.selected) {
```
Comment (saanobhaai) [inline src/app/project/controllers/collect-records.ctrl.js]: Do we need the conditional here?

File: src/app/project/controllers/collect-records.ctrl.js (hunk 6 above)
Comment (tienhoah) [reply src/app/project/controllers/collect-records.ctrl.js]: if collect record 0 or table control records is 0, it's a falsy, the filter label won't display anything.

File: src/app/project/controllers/collect-records.ctrl.js (hunk 1 above)
Comment (tienhoah) [reply src/app/project/controllers/collect-records.ctrl.js]: Removed

File: src/app/project/controllers/collect-records.ctrl.js (hunk 5 above)
Comment (tienhoah) [reply src/app/project/controllers/collect-records.ctrl.js]: Agreed

File: src/app/project/controllers/collect-records.ctrl.js (hunk 7 above)
Comment (tienhoah) [reply src/app/project/controllers/collect-records.ctrl.js]: yeah, only push the ones with the checkbox not selected to local storage

File: src/app/project/controllers/collect-records.ctrl.js (hunk 4 above)
Comment (tienhoah) [reply src/app/project/controllers/collect-records.ctrl.js]: yes I think it's time :)

**Pull request merged**